components/plugins/lints/unrooted_must_root.rs

```diff
@@ -52,11 +52,19 @@ impl LintPass for UnrootedPass {
     }
     /// All structs containing #[must_root] types must be #[must_root] themselves
     fn check_struct_def(&mut self, cx: &Context, def: &ast::StructDef, _i: ast::Ident, _gen: &ast::Generics, id: ast::NodeId) {
-        if cx.tcx.map.expect_item(id).attrs.iter().all(|a| !a.check_name("must_root")) {
-            for ref field in def.fields.iter() {
-                lint_unrooted_ty(cx, &*field.node.ty,
-                                 "Type must be rooted, use #[must_root] on the struct definition to propagate");
+        match cx.tcx.map.get(id) {
+            ast_map::Node::NodeItem(item) => {
+                if item.attrs.iter().all(|a| !a.check_name("must_root")) {
+                    for ref field in def.fields.iter() {
+                        lint_unrooted_ty(cx, &*field.node.ty,
+                        "Type must be rooted, use #[must_root] on the struct definition to propagate");
+                    }
+                }
             }
+            _ => {
+                let parent_id = cx.tcx.map.get_parent(id);
```

Comment:
We should only go up one item here.

See https://github.com/Manishearth/rust-tenacious/commit/e6d19d978698a8913f7c5f9a71670756c3add595